### PR TITLE
TCP2 reordered ACK

### PR DESCRIPTION
**Pull request by ozhuraki**
These patches fix the issue with the data packet being false
identified as a final ACK in case the final ACK
for the connection establishment arrives out of order.

There are a few other less significant fixes and refactorings.

### CODE REVIEW

File: subsys/net/ip/tcp2.c [hunk 1]
```diff
@@ -1435,17 +1435,25 @@ struct net_tcp_hdr *net_tcp_input(struct net_pkt *pkt,
 static sys_slist_t tp_q = SYS_SLIST_STATIC_INIT(&tp_q);
```
Comment (jukkar) [inline subsys/net/ip/tcp2.c]: The commit subject and body has same information in commit "net: tcp2: Support arbitrary requests in tcp_win_pop()" and couple of others too. I would suggest that this is not done like this, the subject in this case is fine, but the body of the commit message should answer the question "why this change is done" and it is not doing it now.

File: subsys/net/ip/tcp2.c [hunk 2]
```diff
@@ -1493,7 +1493,7 @@ static struct net_buf *tcp_win_pop(struct tcp_win *w, const char *name,
 	return out;
```
Comment (jukkar) [inline subsys/net/ip/tcp2.c]: In commit "net: tp: Rename tcp_recv() into tp_tcp_recv()" the subject and body contain same information. Better to say in body something like "Because of ...xyz..., rename tcp_recv() into tp_tcp_recv()" or something like that which gives more information.

**Review by jukkar — CHANGES_REQUESTED** [review]
Lot of commits in this PR, will continue later. As a general rule, the commit body should answer why the commit exists and it is not like that in many of the commits in this PR.

**Comment by ozhuraki** [comment]
@jukkar 

The commit messages are updated, please take a look.

**Review by jukkar — APPROVED** [review]
